src/com/cdd/bao/template/SchemaVocab.java

```diff
@@ -187,16 +187,10 @@ public static SchemaVocab deserialise(InputStream istr, Schema[] templates) thro
 			
 			int numAltLabels = data.readInt();
 			if (numAltLabels > 0)
-			{
-				term.altLabels = new String[numAltLabels];
-				for (int i = 0; i < numAltLabels; i++) term.altLabels[i] = data.readUTF();
-			}
+				for (int i = 0; i < numAltLabels; i++) term.altLabels.add(data.readUTF());
```

Comment:
Here is a comment I made before your second commit.

In the old code, altLabels was newly allocated, which means it was empty before the loop. To be on the save side, add a term.altLabels.clear() before the loop.